packages/shared/src/db/queries.ts

```diff
@@ -2151,7 +2210,60 @@ export const insertLatestPosts = createWriteQuery(
   ['posts']
 );
 
+export const insertUnconfirmedPosts = createWriteQuery(
+  'insertUnconfirmedPosts',
+  async ({ posts }: { posts: Post[] }, ctx: QueryCtx) => {
+    if (!posts.length) {
+      return;
+    }
+    if (posts.some((p) => p.syncedAt != null)) {
+      throw new Error('insertUnconfirmedPosts: posts should not have syncedAt');
+    }
+    return withTransactionCtx(ctx, async (txCtx) => {
+      // insertPosts does multiple queries internally, so we need to wrap it in a transaction
+      await insertPosts(posts, txCtx);
+    });
+  },
+  ['posts']
+);
+
 async function insertPosts(posts: Post[], ctx: QueryCtx) {
+  // HACK: I can't get onConflictDoUpdate to work - manually manage conflicts.
```

Comment:
any ideas? even with `onConflictDoUpdate` configured in a pretty simple way, I had `UNIQUE` constraint errors when replacing the unconfirmed posts with confirmed ones.

I think the code was like

```ts
      .onConflictDoUpdate({
        target: $posts.id
        set: conflictUpdateSetAll($posts),
        // with or without
        setWhere: sql`excluded.synced_at IS NOT NULL OR synced_at IS NULL`
      });
```

Reply:
I'm not sure what the best solution is, but the likely issue is that afaict drizzle doesn't support multiple `onConflictDoUpdate` clauses (open issue here: https://github.com/drizzle-team/drizzle-orm/issues/2276)

Reply:
very sad, added reference to that issue inline: [Add Drizzle issue reference](https://github.com/tloncorp/tlon-apps/pull/4280/commits/db48efb5607057185addc58694ff57d81b9b1410)